user-interface/src/data-verification/ConsolidationOrderAccordion.test.tsx

```diff
@@ -171,51 +259,206 @@ describe('ConsolidationOrderAccordion tests', () => {
   test('should correctly enable/disable buttons', async () => {
     renderWithProps();
     openAccordion(order.id!);
+    setupApiGetMock({ bCase: order.childCases[0] });
 
-    const checkbox: HTMLInputElement = screen.getByTestId(
+    const firstCheckbox: HTMLInputElement = screen.getByTestId(
       `checkbox-case-selection-${order.id}-case-list-0`,
     );
 
-    const includeAllButton = document.querySelector(`.checkbox-toggle label`);
+    const includeAllCheckbox = document.querySelector(`.checkbox-toggle label`);
+    const approveButton = findApproveButton(order.id!);
+    const rejectButton = findRejectButton(order.id!);
+
+    selectTypeAndMarkLead();
 
-    const approveButton = document.querySelector(`#accordion-approve-button-${order.id}`);
     expect(approveButton).not.toBeEnabled();
+    expect(rejectButton).not.toBeEnabled();
 
-    fireEvent.click(checkbox);
+    fireEvent.click(firstCheckbox);
     await waitFor(() => {
       expect(approveButton).toBeEnabled();
+      expect(rejectButton).toBeEnabled();
     });
 
-    fireEvent.click(checkbox);
+    clickMarkLeadButton(0);
     await waitFor(() => {
       expect(approveButton).not.toBeEnabled();
+      expect(rejectButton).not.toBeEnabled();
     });
 
-    const rejectButton = document.querySelector(`#accordion-reject-button-${order.id}`);
-    expect(rejectButton).not.toBeEnabled();
-    fireEvent.click(checkbox);
+    clickMarkLeadButton(0);
     await waitFor(() => {
+      expect(approveButton).toBeEnabled();
       expect(rejectButton).toBeEnabled();
     });
 
-    fireEvent.click(checkbox);
+    fireEvent.click(firstCheckbox);
     await waitFor(() => {
+      expect(approveButton).not.toBeEnabled();
       expect(rejectButton).not.toBeEnabled();
     });
 
-    fireEvent.click(includeAllButton!);
+    fireEvent.click(includeAllCheckbox!);
     await waitFor(() => {
       expect(approveButton).toBeEnabled();
+      expect(rejectButton).toBeEnabled();
     });
 
-    fireEvent.click(includeAllButton!);
+    fireEvent.click(includeAllCheckbox!);
     await waitFor(() => {
       expect(approveButton).not.toBeEnabled();
+      expect(rejectButton).not.toBeEnabled();
     });
 
-    fireEvent.click(checkbox);
+    fireEvent.click(firstCheckbox);
+    await waitFor(() => {
+      expect(approveButton).toBeEnabled();
+      expect(rejectButton).toBeEnabled();
+    });
+
+    ///////////// Now testing "case not listed" ///////////////
+
+    const markAsLeadButton = screen.getByTestId('button-assign-lead-0');
+    expect(markAsLeadButton).not.toHaveClass('usa-button--outline');
+
+    // selecting "Lead Case Not Listed" should unselect the "MarkAsLead" button
```

Comment:
Do we need the comments here? In the second case, maybe we could extract a function that makes it clear what we're testing for.

Reply:
extracting a function at this point would be quite messy, parameter-wise.

Reply:
I'll split it into 2 tests.